src/components/Header/Header.js

```diff
@@ -103,66 +120,30 @@ const Header = ({ logout = () => {}, currentUser = undefined }) => {
     </OfflineHide>
   )
 
-  const handleImageError = () => {
-    setHasImageError(true)
-  }
-
-  const getUserButton = () => {
-    // Avatar with user image
-    if (currentUser && currentUser.picture && !hasImageError) {
-      return (
-        <AvatarWrapper>
-          <CurrentUserImg src={currentUser.picture} alt="" onError={handleImageError} />
-        </AvatarWrapper>
-      )
-    }
-
-    // Avatar with fallback image
-    if (currentUser && currentUser.picture && hasImageError) {
-      return (
-        <AvatarWrapper>
-          <BiggerIconUser />
-        </AvatarWrapper>
-      )
-    }
-
-    // First name
-    if (currentUser && currentUser.first_name) {
-      return (
-        <AvatarWrapper>
-          {currentUser && currentUser.first_name} <IconDown />
-        </AvatarWrapper>
-      )
-    }
-
-    // Full name
-    if (currentUser && currentUser.full_name) {
-      return (
-        <AvatarWrapper>
-          {currentUser && currentUser.full_name} <IconDown />
-        </AvatarWrapper>
-      )
-    }
-
-    // User icon
-    return (
+  const profileImageButton = isProfileImageButtonShowing ? (
+    <AvatarWrapper>
+      <CurrentUserImg src={currentUser?.picture} alt="" onError={handleImageError} />
+    </AvatarWrapper>
+  ) : null
+  const initialsButton =
+    isInitialsButtonShowing && !isProfileImageButtonShowing ? (
       <AvatarWrapper>
-        <BiggerIconUser />
+        <UserCircle>
+          {currentUserFirstInitial}
+          {currentUserLastInitial}
+        </UserCircle>
       </AvatarWrapper>
-    )
-  }
+    ) : null
 
-  const getUserDisplayName = () => {
-    if (currentUser && currentUser.first_name) {
-      return currentUser.first_name
-    }
+  const fallbackButton = (
+    <AvatarWrapper>
+      <BiggerIconUser />
+    </AvatarWrapper>
+  )
 
-    if (currentUser && currentUser.first_name) {
-      return currentUser.full_name
-    }
+  const userIconButton = profileImageButton || initialsButton || fallbackButton
```

Comment:
Ill abstract this into a user icon component in upcoming work. Turns out similar issues happen in the users table too.